scene/main/missing_node.h

```diff
@@ -39,12 +39,17 @@ class MissingNode : public Node {
 	String original_class;
 	String original_scene;
 	bool recording_properties = false;
+	bool recording_signals = false;
 
 protected:
 	bool _set(const StringName &p_name, const Variant &p_value);
 	bool _get(const StringName &p_name, Variant &r_ret) const;
 	void _get_property_list(List<PropertyInfo> *p_list) const;
 
+#ifdef DEBUG_ENABLED
+	virtual Error connect(const StringName &p_signal, const Callable &p_callable, uint32_t p_flags = 0) override;
+#endif
```

Comment:
I was wondering why this is `DEBUG_ENABLED`, when the rest of this class isn't guarded similarly.

I found that `Object::connect` is defined like this:
```cpp
// When in debug, some non-virtual functions can be overridden for multithreaded guards.
#ifdef DEBUG_ENABLED
#define MTVIRTUAL virtual
#else
#define MTVIRTUAL
#endif // DEBUG_ENABLED
```
```cpp
        MTVIRTUAL Error connect(const StringName &p_signal, const Callable &p_callable, uint32_t p_flags = 0);
```

So we're kind of abusing the system here by using `connect` as a virtual method. It seems it's not meant to be, and was just made virtual for debug builds to add thread guards.

Maybe it's fine to (ab)use the system this way for something that's also fairly specific to debug/editor functionality.

Reply:
I don't recall the details, but yeah committing a crime against core architecture for the sake of warnings is a not great. I assume this was also done in a more frequently used node or resource type which could have an impact on performance of connecting signals.

Here, this is done on MissingNode which is only used in case of reloading extensions in the editor, so it's not a hot path. And more importantly, the purpose is to allow converting C# scripts from the `mono` module to C# classes compatible with the `dotnet` module, so the stakes are different :)

It's still a hack, but not all hacks are created equal.